Sources/Flows/OAuth2.swift

```diff
@@ -391,6 +391,9 @@ open class OAuth2: OAuth2Base {
 					callback(json, nil)
 				}
 				catch let error {
+					// Fixes [Issue #367](https://github.com/p2/OAuth2/issues/367)
+					// Refresh token needs to be cleared out upon error, otherwise re-authorizing will not ocurr because the library thinks it has a valid refresh token and tries to fetch a new access token with an expired refresh token.
+					self.clientConfig.refreshToken = nil
```

Comment:
This would clear the refresh token on any error, also e.g. when the server is temporarily unavailable (`OAuth2Error.temporarilyUnavailable`). I don't think this is desirable. The right way would be to check if the error is the `invalid_grant` error (which is thrown as `OAuth2Error.invalidGrant`) and delete the refresh token only in that case.